src/test/java/org/folio/rest/impl/ValidatorRegistryTest.java

```diff
@@ -0,0 +1,484 @@
+
+  private static final JsonObject REGEXP_RULE_ENABLED = new JsonObject()
+    .put("name", "regexp rule enabled")
+    .put("type", "RegExp")
+    .put("validationType", "Strong")
+    .put("state", "Enabled")
+    .put("moduleName", "mod-password-validator")
+    .put("expression", "^(?=.*[A-Za-z])(?=.*\\d)[A-Za-z\\d]+$")
+    .put("description", "At least one letter and one number")
+    .put("orderNo", 1)
+    .put("errMessageId", "");
+
+  private static final JsonObject REGEXP_RULE_DISABLED = new JsonObject()
+    .put("name", "regexp rule disabled")
+    .put("type", "RegExp")
+    .put("validationType", "Strong")
+    .put("state", "Disabled")
+    .put("moduleName", "mod-password-validator")
+    .put("expression", "^(?=.*[A-Za-z])(?=.*\\d)[A-Za-z\\d]+$")
+    .put("description", "At least one letter and one number")
+    .put("orderNo", 1)
+    .put("errMessageId", "");
+
+  private static final JsonObject INVALID_RULE_NEGATIVE_ORDER_NUMBER = new JsonObject()
+    .put("name", "invalid rule with negative order number")
+    .put("type", "Programmatic")
+    .put("validationType", "Soft")
+    .put("state", "Disabled")
+    .put("moduleName", "mod-password-validator")
+    .put("expression", "")
+    .put("implementationReference", "Some implementation")
+    .put("description", "Programmatic rule")
+    .put("orderNo", -1)
+    .put("errMessageId", "");
+
+  private static final JsonObject INVALID_REGEXP_RULE_SOFT = new JsonObject()
+    .put("name", "invalid regexp rule with soft validation type")
+    .put("type", "RegExp")
+    .put("validationType", "Soft")
+    .put("state", "Enabled")
+    .put("moduleName", "mod-password-validator")
+    .put("expression", "^(?=.*[A-Za-z])(?=.*\\d)[A-Za-z\\d]+$")
+    .put("description", "At least one letter and one number")
+    .put("orderNo", 1)
+    .put("errMessageId", "");
+
+  private static final JsonObject VALID_RULE = new JsonObject()
+    .put("name", "valid regexp rule")
+    .put("type", "RegExp")
+    .put("validationType", "Strong")
+    .put("state", "Enabled")
+    .put("moduleName", "mod-password-validator")
+    .put("expression", "^(?=.*[A-Za-z])(?=.*\\d)[A-Za-z\\d]+$")
+    .put("description", "At least one letter and one number")
+    .put("orderNo", 1);
+
+  private static final String TENANT_RULES_PATH = "/tenant/rules";
+  private static final String HOST = "http://localhost:";
+  private static final String HTTP_PORT = "http.port";
+  private static final String TENANT = "diku";
+  private static final String RULE_ID = "ruleId";
+  private static final String VALIDATION_RULES_TABLE_NAME = "validation_rules";
+
+  private static Vertx vertx;
+  private static int port;
+
+  @Rule
+  public Timeout rule = Timeout.seconds(180);
+
+  @BeforeClass
+  public static void setUpClass(final TestContext context) throws IOException {
+    Async async = context.async();
+    vertx = Vertx.vertx();
+    port = NetworkUtils.nextFreePort();
+
+    PostgresClient.setIsEmbedded(true);
```

Comment:
Consider adding the option to use an existing postgres instance because starting embedded postgres is slow.
See https://github.com/folio-org/mod-inventory-storage/blob/0fda3ab/src/test/java/org/folio/rest/api/StorageTestSuite.java#L70-L89 for an example.